prov/util/src/util_mr_monitor.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2017 Cray Inc. All rights reserved.
+ * Copyright (c) 2017 Intel Inc. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <ofi_mr.h>
+
+int ofi_mr_monitor_init(struct ofi_mr_monitor *mr_monitor)
+{
+	/* TODO open Notifier, set subscribe/unsubscribe/get_event here */
+	fastlock_init(&mr_monitor->lock);
+	dlist_init(&mr_monitor->eq_list);
+	return FI_SUCCESS;
+}
+
+void ofi_mr_monitor_finalize(struct ofi_mr_monitor *mr_monitor)
+{
+	/* TODO close Notifier here */
+	assert(dlist_empty(&mr_monitor->eq_list));
+}
```

Comment:
The init/finalize calls are fine, but regarding the comment, I was thinking more along the lines of this sort of use case:
```
struct gni_monitor {
    struct ofi_mr_monitor monitor;
    ...
};

int gni_monitor_init(struct gni_monitor *gni_monitor)
{
    ofi_mr_monitor_init(&gni_monitor.monitor);
    ...
}

void gni_monitor_finalize(struct gni_monitor *gni_monitor)
{
   ...
   ofi_mr_monitor_finalize(&gni_monitor.monitor);
}
```